### PR TITLE
Round sky coordinates to arcseconds for the SHA calculation.

### PR DESCRIPTION
**Pull request by CraigLoomis**

### CODE REVIEW

File: python/pfs/datamodel/utils.py [hunk 1]
```diff
@@ -30,7 +30,6 @@ def calculatePfsVisitHash(visits):
         raise ValueError(f"List of visits is not unique: {[vv for vv in counts if counts[vv] > 1]}")
     return createHash([str(vv).encode() for vv in sorted(visits)])
 
-
```
Comment (PaulPrice) [inline python/pfs/datamodel/utils.py:33]: Need to keep two lines between functions to satisfy linter.

File: python/pfs/datamodel/utils.py [hunk 2]
```diff
@@ -45,8 +44,13 @@ def calculate_pfsDesignId(fiberIds, ras, decs):
     if (ras == 0.0).all() and (decs == 0.0).all():  # don't check fiberIds as this may be lab data
         return 0x0
 
-    return createHash(["%d %.0f %.0f" % (fiberId, ra, dec) for fiberId, ra, dec in zip(fiberIds, ras, decs)])
+    def _roundToArcsec(d):
+        return int(d*3600.0 + 0.5)/3600.0
 
+    # Regardless of the arcsec rounding, we need to choose a precision for the string respresentation.
+    # If datamodel.txt phrasing were a little sloppier, we could just use integer arcseconds.
```
Comment (PaulPrice) [inline python/pfs/datamodel/utils.py:51]: Changing the wording of datamodel.txt is an option...?

File: python/pfs/datamodel/utils.py [hunk 3]
```diff
@@ -45,8 +44,13 @@ def calculate_pfsDesignId(fiberIds, ras, decs):
...
+    return createHash(["%d %0.6f %0.6f" % (fiberId, _roundToArcsec(ra), _roundToArcsec(dec))
+                       for fiberId, ra, dec in zip(fiberIds, ras, decs)])
 
 def createHash(*args):
```
Comment (PaulPrice) [inline python/pfs/datamodel/utils.py:55]: Is there a reason we have this separate implementation of `pfs.datamodel.utils.createHash`?

**Comment by alefur** [comment]
This imply a new sha for pfsDesign, which I already need to regenerate anyway (eg scienceFiberIds - fiberIds confusion.)
Patching the headers is a bit painful, so if we can hit two birds with one stone I would appreciate it.

@PaulPrice what do you think ?

**Review by PaulPrice — APPROVED** [review]

**Comment by PaulPrice** [comment]
I don't know what you were doing with the top-end during this run, but if there were only a few settings of interest then I wonder if it might be a good idea to introduce additional magic `pfsDesignId` values (which are the values we already have, so we don't have to bother fixing the headers). We already have some: `0x0` means lab data, we have all the dummy cable B values, and `0xdeadbeef` means SuNSS. A few more wouldn't hurt...

**Comment by alefur** [comment]
Well, we're testing different cobra configuration eg different x,y (pfiNominal in the pfsDesign).
But we don't have real ra and dec for those, and we still need a unique sha.
As pfsDesignId is calculated from ra,dec,fiberId, We (Robert) artificially set ra and dec to. 
ra = 100 + 1e-3*x
dec = 100 + 1e-3*y

File: python/pfs/datamodel/utils.py (hunk 1 above)
Comment (CraigLoomis) [reply python/pfs/datamodel/utils.py]: Know that now. OK.

File: python/pfs/datamodel/utils.py (hunk 3 above)
Comment (CraigLoomis) [reply python/pfs/datamodel/utils.py:57]: Not sure I follow. `createHash`, a.k.a. `pfs_datamodel.utils.createHash` takes strings and returns a hash. `calculate_pfsDesignId` generates the strings.

**Comment by PaulPrice** [comment]
I don't think that it's important that we have `pfsDesignId` equal to a particular hash. What's important is that there is a `pfsDesignId` and that it's unique. We have a `pfsDesignId` and it's unique. So even if it's the wrong hash, I think we can just keep it, update the content of the files and not update the `pfsDesignId`. I don't think there's anything in the `PfsDesign` code that will change the `pfsDesignId` out from under us, so it should work. No header modifications needed.

File: python/pfs/datamodel/utils.py (hunk 3 above)
Comment (PaulPrice) [reply python/pfs/datamodel/utils.py:57]: But the `createHash` function has been copied in wholesale from `pfs.datamodel.utils`, and that's what `calculate_pfsDesignId` is calling when it should be calling the implementation in `pfs.datamodel.utils`.

**Comment by alefur** [comment]
Fair enough.
keep the same pfsDesignId for the existing one, but correct the content (eg fiberId)
Only the newly generated pfsDesign will have the consistent hash (fiberId,ra,dec)

File: python/pfs/datamodel/utils.py (hunk 3 above)
Comment (CraigLoomis) [reply python/pfs/datamodel/utils.py:57]: ??? OK, now I am really confused. This IS `python/pfs/datamodel/utils.py`; the `createHash` within that is what is being called, and has not been copied anywhere.

Comment (PaulPrice) [reply python/pfs/datamodel/utils.py:57]: Oh. Then I'm just stupid. Sorry!

**Pull request merged**